force-app/main/default/classes/RD2_UpdateCommitmentBulkService.cls

```diff
@@ -0,0 +1,159 @@
+                requestService = new UTIL_Http.RequestService();
+            }
+            return requestService;
+        }
+        set;
+    }
+
+    private List<Commitment> commitments;
+
+    public RD2_UpdateCommitmentBulkService withCommitments(List<npe03__Recurring_Donation__c> rds) {
+        if (rds == null || rds.isEmpty()) {
+            return this;
+        }
+
+        this.commitments = new List<Commitment>();
+        for (npe03__Recurring_Donation__c rd : rds) {
+            this.commitments.add(new Commitment(rd));
+        }
+
+        return this;
+    }
+
+    public void processCommitments() {
+        if (!RD2_ElevateIntegrationService.isIntegrationEnabled() ) {
+            return;
+        }
+
+        for (Integer i = 0; i < MAXIMUM_API_CALL_PER_TRANSACTION; i++) {
+            if (noCommitmentToProcess()) {
+                break;
+            }
+
+            RequestBody reqBody = new RequestBody();
+            reqBody.commitments =  getCommitmentsToProcess();
+
+            HttpRequest request = buildBulkStatusUpdateRequest(reqBody);
+            UTIL_Http.Response response;
+            try {
+                response = requestService.sendRequest(request);
+            } catch (Exception ex) {
+                response = requestService.buildErrorResponse(ex);
+            }
+
+            processResponse(response);
+        }
+
+        this.logger.processErrors();
+    }
+
+    private List<Commitment> getCommitmentsToProcess() {
+        List<Commitment> toProcess = new List<Commitment>();
+
+        for (Integer i = 0; i < REQUEST_SIZE; i++) {
+            if (noCommitmentToProcess()) {
+                break;
+            }
+
+            toProcess.add(this.commitments.remove(0));
+        }
+        return toProcess;
+    }
+
+    private Boolean noCommitmentToProcess() {
+        return this.commitments == null || this.commitments.isEmpty();
+    }
+
+    private HttpRequest buildBulkStatusUpdateRequest(RequestBody reqBody) {
+        return new PS_Request.Builder()
+            .withMethod(UTIL_Http.Method.POST)
+            .withEndpoint(PS_Request.ElevateEndpoint.COMMITMENT_UPDATE_BULK)
+            .withRecommendedTimeout()
+            .withBody(JSON.serialize(reqBody))
+            .build();
+    }
+
+    private void processResponse(UTIL_Http.Response response) {
+        if (isCommitmentSuccess(response)) {
+            return;
+        }
+
+        this.logger.addError(
+            null,
+            response.getErrorMessages(),
+            RD2_ElevateIntegrationService.LOG_TYPE_COMMITMENT_UPDATE_BULK
+        );
+    }
+
+    private Boolean isCommitmentSuccess(UTIL_Http.Response response) {
+        return response.statusCode == UTIL_Http.STATUS_CODE_CREATED
+            || response.statusCode == UTIL_Http.STATUS_CODE_OK;
+    }
+
+    public class RequestBody {
+        @TestVisible
+        List<Commitment> commitments;
+    }
+
+    public class Commitment {
```

Comment:
Could we extract this out into its own class? I can see this being used again that we could continue building additional behavior around as its own entity